### PR TITLE
Access `replace_functions_map` via PreParforPass instance

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
`replace_functions_map` is global variable from `numba.parfors.parfor`.
It is necessary to have possibility to modify `replace_functions_map`
depending on PreParforPass instance and access to it via the instance.

This PR is a proposal for starting a discussion.

### CODE REVIEW

**Comment by sklam** [comment]
This change will very likely be needed. I cannot think of any better way to specialize the parfor passes.

**Comment by stuartarchibald** [comment]
From out of band discussion, something along the lines of this patch is needed and moving information out of globals is a good idea. Once tests are added this PR can be reviewed. Many thanks.

File: numba/parfors/parfor.py [hunk 1]
```diff
@@ -1350,14 +1350,15 @@ class PreParforPass(object):
     implementations of numpy functions if available.
     """
     def __init__(self, func_ir, typemap, calltypes, typingctx, options,
-                 swapped={}):
+                 swapped={}, replace_functions_map=replace_functions_map):
```
Comment (sklam) [inline numba/parfors/parfor.py:1353]: I'd suggest letting it defaulting to `None` and setting to the `replace_functions_map` global dictionary in the function body. With that, the test do not need to know what should be the default dict.

File: numba/tests/test_parfors_passes.py [hunk 2]
```diff
@@ -73,13 +73,16 @@ def _run_parfor(cls, test_func, args):
 
             diagnostics = numba.parfors.parfor.ParforDiagnostics()
 
+            if swap_map is None:
+                swap_map = numba.parfors.parfor.replace_functions_map
```
Comment (sklam) [inline numba/tests/test_parfors_passes.py:77]: the test should not need to know the default value. see previous comment.

**Review by sklam — COMMENTED** [review]
Thanks for the patch. I have a few suggestion.

**Review by sklam — APPROVED** [review]
LGTM. I restarted CI since it was failing due to internet traffic jam. Can merge once CI reports ok.

**Comment by sklam** [comment]
The 404 error in CI failure did not recover and it is unrelated to this PR. Marking this for merge.

**Pull request merged**